### PR TITLE
Migrate Convert Block Type test to Playwright

### PR DESCRIPTION
**Pull request by pooja-muchandikar**
## What?

Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate [convert-block-type.test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/convert-block-type.test.js) to its Playwright version.
## Why?

Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions

Run` npm run test-e2e:playwright test/e2e/specs/editor/various/convert-block-type.spec.js`

### CODE REVIEW

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726, @talldan 👋 

I hope you are doing great!

I have raised another migration PR, please could anyone of you help me in reviewing this PR?

Thanks!!

File: test/e2e/specs/editor/various/convert-block-type.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		page,
+		editor,
+	} ) => {
+		const code = 'print "Hello Dolly!"';
+
+		await editor.insertBlock( { name: 'core/code' } );
+		await page.keyboard.type( code );
+
+		// Verify the content starts out as a Code block.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/convert-block-type.spec.js]: Nit: Probably will be better to get rid of the snapshot and just inline the expected result here:

```suggestion
		await expect.poll( editor.getEditedPostContent )
			.toBe( `<!-- wp:code -->
<pre class="wp-block-code"><code>${code}</code></pre>
<!-- /wp:code -->` );
```

File: test/e2e/specs/editor/various/convert-block-type.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+		await editor.transformBlockTo( 'core/preformatted' );
+
+		// The content should now be a Preformatted block with no data loss.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/convert-block-type.spec.js]: Nit: Same here.

File: test/e2e/specs/editor/various/convert-block-type.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Code block', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/convert-block-type.spec.js:8]: I think we need to remember to `deleteAllPosts()` in `test.afterEach` if we create one in `test.beforeEach` 😅 (I also forgot this all the time).

**Review by kevin940726 — APPROVED** [review]
Thanks for the PR! Left some nitpicks but they're optional, I think this looks good!

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

I have addressed all the feedbacks please could you check once?

**Comment by kevin940726** [comment]
Awesome work! Thanks!

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

Looks like the PR details are not added here- https://github.com/WordPress/gutenberg/issues/38851

**Comment by kevin940726** [comment]
Added! Thanks for letting me know!

**Comment by oandregal** [comment]
Added label for the changelog.

**Pull request merged**